src/adapter/cloudflare-pages/handler.ts

```diff
@@ -34,6 +43,34 @@ export const handle: HandleInterface = (app: Hono) => (eventContext: EventContex
   )
 }
 
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function handleMiddleware<E extends Env = any, P extends string = any, I extends Input = {}>(
+  middleware: MiddlewareHandler<E, P, I>
+): PagesFunction {
+  return async (executionCtx) => {
+    const context = new Context(executionCtx.request, {
+      env: executionCtx.env,
+      executionCtx,
+    })
+
+    let response: Response | void = undefined
+
+    try {
+      response = await middleware(context, async () => {
+        context.res = await executionCtx.next()
+      })
+    } catch (error) {
+      if (error instanceof HTTPException) {
+        response = error.getResponse()
+      } else {
+        throw error
+      }
```

Comment:
I included the code to handle `HTTPException`, but I wasn't sure what should happen with other kinds of errors, so it will re-throw everything else

Looking at the documentation, there is also a [`Context.error`](https://hono.dev/docs/api/context#error) property. Should we handle that here too? Similar to `Context.res`

Reply:
You are right. It should be re-thrown. Thanks.

> Looking at the documentation, there is also a [`Context.error`](https://hono.dev/docs/api/context#error) property. Should we handle that here too?

Yes. Please put the error to `context.error`.